### ISSUE TITLE
Use of "code blocks" for event handlers can lead to buggy code

### ISSUE THREAD

**Issue by jpsacha** [scalafx/scalafx#102]
_From [jpsacha](https://code.google.com/u/jpsacha/) on October 25, 2013 17:20:14_

Currently there is an implicit conversion that turns a code block, and pretty much anything into an event handler. This is can easy lead to buggy code.

The problem with code blocks is that they are not interpreted as functions in Scala, only the last statement is. Consider following code, (the code taken from scalafx.animation.KeyFrameSpec:

```
var callCount1 = 0
var callCount2 = 0
val finishHandler = {
  callCount1 += 1
  callCount2 += 1
}
```

Now let use `finishHandler` three times :

```
KeyFrame(10 ms, onFinished = finishHandler).onFinished.handle(null)
KeyFrame(10 ms, onFinished = finishHandler).onFinished.handle(null)
KeyFrame(10 ms, onFinished = finishHandler).onFinished.handle(null)
```

At this point have:
   callCount1 == 1
and
   callCount2 == 3  

What happened? The body of `finishHandler` is a code block, not a function. Scala is a bit strange about code blocks, it will treat everything but the last line as a constructor (it will be executed once and hence he have callCount1 == 1), but the last statement will be interpreted as a function, so we will have callCount2 == 3.

Basically there should be no implicit conversions that turn a code block into an event handler, as this may lead to difficult to understand and buggy code. In particular `scalafx.event.EventIncludes#eventClosureWrapper`:

implicit def eventClosureWrapper[T <: jfxe.Event](handler: => Any) = new jfxe.EventHandler[T] { ... }

should be removed. This will require some changes in animation API that is relaying on it.

_Original issue: http://code.google.com/p/scalafx/issues/detail?id=102_
